models/meteora/meteora_trades.sql

```diff
@@ -0,0 +1,159 @@
+            , sp.call_tx_signer as trader_id
+            , sp.call_tx_id as tx_id
+            , sp.call_outer_instruction_index as outer_instruction_index
+            , COALESCE(sp.call_inner_instruction_index, 0) as inner_instruction_index
+            , sp.call_tx_index as tx_index
+            , COALESCE(tk_2.token_mint_address, cast(null as varchar)) as token_bought_mint_address
+            , COALESCE(tk_1.token_mint_address, cast(null as varchar)) as token_sold_mint_address
+            , trs_2.account_source as token_bought_vault
+            , trs_1.account_destination as token_sold_vault
+        FROM (
+            SELECT 
+                sp.*
+                , dp.call_inner_instruction_index as deposit_index
+                , row_number() over (partition by sp.call_tx_id, sp.call_outer_instruction_index, sp.call_inner_instruction_index order by dp.call_inner_instruction_index asc) as first_deposit
+            FROM {{ source('meteora_pools_solana', 'amm_call_swap') }} sp
+            LEFT JOIN {{ source('meteora_vault_solana', 'vault_call_deposit') }} dp ON sp.call_tx_id = dp.call_tx_id 
+                AND sp.call_block_slot = dp.call_block_slot
+                AND sp.call_outer_instruction_index = dp.call_outer_instruction_index 
+                and COALESCE(sp.call_inner_instruction_index, 0) < dp.call_inner_instruction_index
+        ) sp
+        INNER JOIN {{ source('spl_token_solana', 'spl_token_call_transfer') }} trs_1 
+            ON trs_1.call_tx_id = sp.call_tx_id 
+            AND trs_1.call_block_time = sp.call_block_time
+            AND trs_1.call_outer_instruction_index = sp.call_outer_instruction_index 
+            AND trs_1.call_inner_instruction_index = sp.deposit_index + 1
+            {% if is_incremental() %}
+            AND {{incremental_predicate('trs_1.call_block_time')}}
+            {% else %}
+            AND trs_1.call_block_time >= TIMESTAMP '{{project_start_date}}'
+            {% endif %}
+        INNER JOIN {{ source('spl_token_solana', 'spl_token_call_transfer') }} trs_2 
+            ON trs_2.call_tx_id = sp.call_tx_id 
+            AND trs_2.call_block_time = sp.call_block_time
+            AND trs_2.call_outer_instruction_index = sp.call_outer_instruction_index 
+            AND trs_2.call_inner_instruction_index = sp.deposit_index + 4
+            {% if is_incremental() %}
+            AND {{incremental_predicate('trs_2.call_block_time')}}
+            {% else %}
+            AND trs_2.call_block_time >= TIMESTAMP '{{project_start_date}}'
+            {% endif %}
+        --we want to get what token was transfered out first as this is the sold token. THIS MUST BE THE DESTINATION account, the source account is commonly created/closed through swap legs.
+        LEFT JOIN {{ ref('solana_utils_token_accounts') }} tk_1 ON tk_1.address = trs_1.account_destination
+        LEFT JOIN {{ ref('solana_utils_token_accounts') }} tk_2 ON tk_2.address = trs_2.account_source
```

Comment:
didn't we make this spell incremental? if so, apply the usual time filter on these join conditions too